File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
    const answer = [];
    
    queries.forEach((query) => {
        const [s, e, k] = query;
        const mins = arr.slice(s, e + 1).filter(el => el > k);
        mins.length === 0 ?
            answer.push(-1) :
            answer.push(Math.min(...mins));
    });
    
    return answer;
}

File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        List<Integer> result = new ArrayList<>();
        
        for (int[] query : queries) {
            int s = query[0], e = query[1], k = query[2];
            int min = 1000001;
            int[] copy = Arrays.copyOfRange(arr, s, e + 1);
            int[] filter = Arrays.stream(copy).filter(num -> num > k).toArray();
            
            for (int num : filter) {
                if (min > num) {
                    min = num;
                }
            }
            
            if (min == 1000001) {
                result.add(-1);
            } else {
                result.add(min);
            }
        }
        
        return result.stream().mapToInt(Integer::intValue).toArray();
    }
}